### DISCUSSION TITLE
Multiple Edumeet Room Servers with load balancer

### DISCUSSION THREAD

**Discussion by Alex-YddeR** [edumeet/edumeet-room-server#24] [Deployment]
Dear Team,

I have been testing by deploying multiple room servers with load balancer, however when users connected they get into different sessions but not in same session.
I don't see configuration for redis in room server to manage this

Thanks
Alex

**Comment by havfo** [comment]
There is no communication between room-servers, so you need to make sure that all users in the same room end up on the same room-server. This can be done with ex. HAProxy stick tables or similar..

**Comment by Alex-YddeR** [comment]
ok, how about media-nodes can different users can connect to different media-node and can be same session. Because when i tested i see multiple users connected to different media-node and they are in same session.

**Comment by havfo** [comment]
That is correct, any user can be on any media-node and media will be piped between them. A note there though, at the moment, the load balancing between media-node is not done yet, and is simply round-robin.

**Comment by pnts-se** [comment]
@Alex-YddeR I am in the process of setting up multiple room-servers with HAProxy in front, and will be facing the same challenges. Perhaps we could convert this to a discussion and share our findings and results?

**Comment by havfo** [comment]
An example from Jitsi: https://community.jitsi.org/t/haproxy-configuration/58009

**Comment by pnts-se** [comment]
@Alex-YddeR I will start to look at this during next week. Haproxy in front doing SSL termination, and multiple room-server instances in Kubernetes behind.

**Comment by pnts-se** [comment]
With this haproxy configuration, peers joining the same room end up on the same room-server:

```
peers edumeet-peers
    peer lb01 192.168.1.1:12345
    peer lb02 192.168.1.2:12345

backend room_servers
        balance roundrobin
        stick-table type string len 256 size 200k expire 180m peers edumeet-peers
        stick on urlp(roomId)
        option forwardfor
        server worker1 10.0.0.1:30201 check
        server worker1 10.0.0.1:30202 check
```

**Reply by Alex-YddeR** [reply]
Hello Pontus Fagerström,

Can you please help me with the HA proxy configuration for room servers, we have client installed with AWS load balancer. 
Our room-server is getting hanged and not restoring back when there are more than 100 connections after some time as below. we are forced to restart the room-server service.


(node:813) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 sessionIdChanged listeners added to [Peer]. Use emitter.setMaxListeners() to increase limit


Thanks
Alex

**Reply by pnts-se** [reply]
@Alex-YddeR with the configuration above peers end up on the same room server, which is a requirement using the current architecture.

That will not solve your problem which seems to be related to exhausting the resource of the room-server. As we haven't focused on scaling out I'm not sure where the bottleneck would be, but in time I'll certainly have a look at the warning you've attached.
